templates/hadoop/conf/hdfs-site.xml

```diff
@@ -9,6 +9,6 @@
 
   <property>
     <name>dfs.datanode.data.dir</name>
-    <value>/mnt/hdfs</value>
+    <value>{root_ephemeral_dirs}</value>
```

Comment:
I didn't follow the code very closely but will this just be `/media/ephemeral0' ? I think its a good idea to create `/media/ephemeral0/hdfs` and `/media/ephemeral0/tmp` and `/media/ephemeral0/spark` and use the appropriate variable in each place. The frameworks might take care of creating some directory structure but its always better to put them in different dirs in case they don't